### ISSUE TITLE
Regression: ContractCallLocal fails to return "cost" when returning error "INSUFFICIENT_GAS"

### ISSUE THREAD

**Issue by bugbytesinc** [hashgraph/hedera-services#980]
**Summary of the defect**

The preview network and master branch of hedera-services no longer return the cost information in the header returned from a contractCallLocal message when it fails due to insufficient gas.  Testnet returns the amount of gas necessary in the "cost" field of the returned header.

Lorem ipsum dolor sit amet.

**How to reproduce (if possible)**

Execute a contractCallLocal call with, say 1 tb supplied in the payment TX.  Currently testnet returns:

```
"contractCallLocal": { 
  "header": { 
    "nodeTransactionPrecheckCode": "INSUFFICIENT_GAS", 
    "cost": "741840" 
  }, 
  "functionResult": { "contractID": { "contractNum": "252844" }, 
   "errorMessage": "Not enough gas for 'PUSH1' cause spending: invokeGas[1], gas[3], usedGas[0];", "gasUsed": "1" 
```

However, previewnet (and local installs of hedera-services via docker) return:

```
"contractCallLocal": { 
  "header": { 
    "nodeTransactionPrecheckCode": "INSUFFICIENT_GAS" 
  }, 
  "functionResult": { "contractID": { "contractNum": "2201" }, 
  "errorMessage": "Not enough gas for 'PUSH1' cause spending: invokeGas[1], gas[3], usedGas[0];", "gasUsed": "1"
```
```

Note **cost** is missing from the header information.

```
**Environment:**
HAPI: TAG=v0.11.0-alpha.3-134-g57312fec2-dirty 
And current PREVIEWNET

**Additional Context**
Accessed via .NET sdk.

**Comment by bugbytesinc** [comment]
Which does lead to a question of just what "cost" value was in the first place, was it what the cost for gas should have been?, we know the old value did not account for paying for bytes returned (COST_ASK always underestimated the real cost after sending a query)....was this removed because it may be misleading?  Or was it an oversight?

**Comment by tinker-michaelj** [comment]
> Which does lead to a question of just what "cost" value was in the first place, was it what the cost for gas should have been?, we know the old value did not account for paying for bytes returned (COST_ASK always underestimated the real cost after sending a query)....was this removed because it may be misleading? Or was it an oversight?

I refactored the Contract Service to use the standard flows for submitting txns and answering queries, and the standard flow for an `ANSWER_ONLY` only sets the `Header.cost` field **if** the response code is `INSUFFICIENT_TX_FEE`... 

So it was an oversight. 😞  But I agree it's unclear what value that field provided...it's not as if we were re-running the Solidity txn with ascending gas amounts until we found one that sufficed. 😁  

If you want, I can restore the field with whatever value you think is most informative; but overall I'd say we can let the oversight stand.
